src/test/java/com/googleintern/wfm/ruleengine/CsvParserTest.java

```diff
@@ -0,0 +1,293 @@
+package src.test.java.com.googleintern.wfm.ruleengine;
+
+import com.opencsv.exceptions.CsvException;
+import org.junit.Assert;
+import org.junit.Test;
+import src.main.java.com.googleintern.wfm.ruleengine.action.CsvParser;
+import src.main.java.com.googleintern.wfm.ruleengine.model.PoolAssignmentModel;
+import src.main.java.com.googleintern.wfm.ruleengine.model.UserPoolAssignmentModel;
+
+import java.io.IOException;
+import java.util.*;
+
+/** * CsvParserTest class is used to testing the functionality of CsvParser class. */
+public class CsvParserTest {
+  private static final String TEST_CSV_FILE_PATH =
+      System.getProperty("user.home")
+          + "/Project/wfm-rule-suggestion-engine/src/"
+          + "test/resources/com/googleintern/wfm/ruleengine/csv_parser_test_data.csv";
+
+  // Expected values of fields for Case 0
+  private static final int CASE_NUMBER_0 = 0;
+  private static final long USER_ID_CASE_0 = 0;
+  private static final long WORKFORCE_ID_CASE_0 = 1024;
+  private static final long WORKGROUP_ID_CASE_0 = 0;
+  private static final List<Long> ROLE_IDS_CASE_0 = new ArrayList<>();
+  private static final List<Long> SKILL_IDS_CASE_0 = new ArrayList<>();
+  private static final List<Long> ROLESKILL_IDS_CASE_0 = new ArrayList<>();
+  private static final Set<PoolAssignmentModel> POOL_ASSIGNMENTS_CASE_0 = new HashSet<>();
+
+  // Expected values of fields for Case 1
+  private static final int CASE_NUMBER_1 = 1;
+  private static final long USER_ID_CASE_1 = 1;
+  private static final long WORKFORCE_ID_CASE_1 = 1024;
+  private static final long WORKGROUP_ID_CASE_1 = 0;
+  private static final List<Long> ROLE_IDS_CASE_1 = Arrays.asList(2020L);
+  private static final List<Long> SKILL_IDS_CASE_1 = new ArrayList<>();
+  private static final List<Long> ROLESKILL_IDS_CASE_1 = new ArrayList<>();
+  private static final Set<PoolAssignmentModel> POOL_ASSIGNMENTS_CASE_1 = new HashSet<>();
+
+  // Expected values of fields for Case 2
+  private static final int CASE_NUMBER_2 = 2;
+  private static final long USER_ID_CASE_2 = 2;
+  private static final long WORKFORCE_ID_CASE_2 = 1024;
+  private static final long WORKGROUP_ID_CASE_2 = 0;
+  private static final List<Long> ROLE_IDS_CASE_2 = Arrays.asList(2020L, 2019L, 2018L);
+  private static final List<Long> SKILL_IDS_CASE_2 = Arrays.asList(2000L);
+  private static final List<Long> ROLESKILL_IDS_CASE_2 = Arrays.asList(1990L);
+  private static final Set<PoolAssignmentModel> POOL_ASSIGNMENTS_CASE_2 =
+      new HashSet<>(
+          Arrays.asList(
+              PoolAssignmentModel.builder().setCasePoolId(2000543).setPermissionSetId(2048).build()
+          )
+      );
```

Comment:
`POOL_ASSIGNMENTS_CASE_2 = ImmutableSet.of(PoolAssignmentModel.builder().setCasePoolId(2000543).setPermissionSetId(2048).build())`